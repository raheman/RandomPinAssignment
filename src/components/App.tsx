import React from 'react';
import '../css/App.css';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import GeneartePin from './GeneartePin';
import 'react-tabs/style/react-tabs.css';
import SavedPins from './SavedPins';

export default class App extends React.Component<any,any> {
  constructor(props: any, context: any) {
      super(props,context);
      this.state={
          generateTab: true
      };
  }

  render(){
      return(
          <div >
            <Tabs >
            <TabList>
              <Tab>Generate</Tab>
              <Tab>Saved</Tab>
            </TabList>
        
            <TabPanel >
              <GeneartePin/>
            </TabPanel>
            <TabPanel>
              <SavedPins/>
            </TabPanel>
          </Tabs>
          </div>
      )
  }
}


