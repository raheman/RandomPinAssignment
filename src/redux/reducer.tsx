import {Reducer} from 'react';
const initialState={
    randomObj:{},
    nameObj:{}

}
const randomPinReducer:Reducer<any,any>=(state=initialState,action:any)=>{
    console.log(action.type);
    switch (action.type){
        case 'SAVE_STATE' :
            var obj = {...state.randomObj}
            var obj1 = {...state.nameObj};
            obj[action.arr[0]] = action.arr;
            obj1[action.arr[0]] = "";
            return {
                ...state,
                 randomObj: obj,
                 nameObj: obj1
            }
        case 'ON_CHANGE_HANDLER':
            var objChange = {...state.nameObj};
            objChange[action.index] = action.name;
            console.log(objChange);
            return {
                ...state,
                nameObj: objChange
            }
        case 'ON_DELETE_HANDLER' : 
            var objDelete = {...state.randomObj}
            var objDelete1 = {...state.nameObj};
            delete objDelete[action.index];
            delete objDelete1[action.index];
            return{
                ...state,
                randomObj: objDelete,
                nameObj: objDelete1
            }
    }

    return state;
}


export default randomPinReducer;
