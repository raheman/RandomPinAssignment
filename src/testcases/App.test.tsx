import React from 'react';
import { shallow, mount } from 'enzyme';
import  App  from '../components/App';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import ReactDOM from 'react-dom';

import reducer from '../redux/reducer';

import {Provider} from 'react-redux';
import {createStore} from 'redux';

const store=createStore(reducer);
configure({ adapter: new EnzymeAdapter() });


describe('App Compomemt Test Cases', () => {
    let wrapper: any;
    let setGeneratedPins = jest.fn();
    wrapper = shallow(<App />)
    test('Snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it("should render without crashing", () => {
        const div = document.createElement("div");
        div.id = "root";
        document.body.appendChild(div);
        require("../index.tsx");
        expect(ReactDOM.render).toBeTruthy();
      });
    
    test('Div Length', () => {
        expect(wrapper.length).not.toBe(null);
    })    
   
    test('Testing Divs Length', () => {
        expect(wrapper.find('div').length).toEqual(1);
    })
    
   
})
