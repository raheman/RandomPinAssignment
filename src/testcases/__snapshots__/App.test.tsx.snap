// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Compomemt Test Cases Snapshot test 1`] = `ShallowWrapper {}`;
